docker/api/image.py

```diff
@@ -342,14 +342,14 @@ def prune_images(self, filters=None):
             params['filters'] = utils.convert_filters(filters)
         return self._result(self._post(url, params=params), True)
 
-    def pull(self, repository, tag=None, stream=False, auth_config=None,
+    def pull(self, repository, tag='latest', stream=False, auth_config=None,
```

Comment:
Thinking out loud: would it work to add an `all_tags` parameter, and based on that either keep `None` as `None`, or set `None -> latest` ?

Train of thought is that if you want to pull _all tags_, no `tag` is allowed to be set, so it feels a bit more natural to having a separate option (the original "no tag means all tags" is a bit awkward, and, well, bad choices in the past)

Reply:
@thaJeztah I added an `all_tags` parameter to the pull methods. Is the logic ok? If we set it, we ignore any tag set with `tag` or `repository` parameter.